data/campaigns/The_South_Guard/scenarios/02_Proven_by_the_Sword.cfg

```diff
@@ -154,26 +154,14 @@
             [/effect]
         [/object]
 
-        [store_unit]
+        [modify_unit]
             [filter]
                 side=2
                 canrecruit=no
                 gender=female
             [/filter]
-            variable=namechange
-        [/store_unit]
-
-        [foreach]
-            array=namechange
-            [do]
-                {VARIABLE this_item.name _"female^Criminal"}
-                [unstore_unit]
-                    variable=this_item
-                    find_vacant=no
-                [/unstore_unit]
-            [/do]
-        [/foreach]
-        {CLEAR_VARIABLE namechange}
+            name= _"female^Criminal"
+        [/modify_unit]
```

Comment:
Instead of doing this, let's deprecate the `NAMED_GENERIC_UNIT` macro and replace it with one that takes both a male and a female name. The macro generates units with a random gender, and with the exception of UtBS S01 (see issue #5632), it's being used with names that might be gendered in non-English languages, so it ought to support `female_name`.

It's only used in TSG and UtBS, with names such as "Criminal", "Westin Guard", "Merman Guard", "Goblin Skirmisher", etc.